android/app/src/main/java/app/eduroam/geteduroam/config/model/EAPIdentityProvider.kt

```diff
@@ -10,29 +10,28 @@ import java.util.Date
 
 @Root(name = "EAPIdentityProvider", strict = false)
 @JsonClass(generateAdapter = true)
-class EAPIdentityProvider {
-
+data class EAPIdentityProvider(
     @field:ElementList(name = "AuthenticationMethods", entry = "AuthenticationMethod")
-    var authenticationMethod: List<AuthenticationMethod>? = null
+    var authenticationMethod: List<AuthenticationMethod>? = null,
```

Comment:
Yes I tried vals, but then the SimpleXML serializer was complaining.
I use data classes for the copy operator, which makes sure that my copy still works correctly if we add extra properties.
About the hash / equals: I don't think it makes a difference compared to the old version, right? There we also did not have an equals / hashcode function, so this would even be an improvement compared to the previous setup.